source/actions/DefaultEventAction.cc

```diff
@@ -86,11 +86,22 @@ REGISTER_CLASS(DefaultEventAction, G4UserEventAction)
 
       G4TrajectoryContainer* tc = event->GetTrajectoryContainer();
       if (tc) {
+        // in interactive mode, a G4TrajectoryContainer would exist
+        // but the trajectories will not cast to Trajectory
+        Trajectory* trj = dynamic_cast<Trajectory*>((*tc)[0]);
+        if (trj == nullptr){
+          G4Exception("[DefaultEventAction]", "EndOfEventAction", FatalException,
+                      "Adequate TrackingAction not registered for DefaultEventAction");
```

Comment:
I would say explicitly which one is the correct action, something like "DefaultTrackingAction is required when using DefaultEventAction".